src/foregroundTimeTracker.ts

```diff
@@ -0,0 +1,100 @@
+export default class ForegroundTimeTracker {
+    private isTrackerActive: boolean = false;
+    private localStorageName: string = '';
+    public startTime: number = 0;
+    public totalTime: number = 0;
+
+    constructor(timerKey: string) {
+        this.localStorageName = `mp-time-${timerKey}`;
+        this.loadTimeFromStorage();
+        this.handleVisibilityChange = this.handleVisibilityChange.bind(this);
+        this.syncAcrossTabs = this.syncAcrossTabs.bind(this);
+        this.init();
+    }
+
+    private init(): void {
+        document.addEventListener(
+            'visibilitychange',
+            this.handleVisibilityChange
+        );
+        window.addEventListener('beforeunload', () => this.updateTimeInPersistence());
+         // Sync time updates across tabs
+        window.addEventListener('storage', this.syncAcrossTabs);
+        this.startTracking();
+
+        // TODO: this is just to ensure when we load it in an app we can see the timer updates in the console
```

Comment:
Sustained Engineering: There could be some value in adding our debug logger here so that if `debug === true`, we can fire this log. I'm fine with it being a ticket for later.

Reply:
i decided to move this ahead, because by calling `getForegroundTime` we end up calling `updateTotalTime` and i just don't want it to create unintended consequences.